frame/system/src/lib.rs

```diff
@@ -395,23 +396,37 @@ pub mod pallet {
 
 		/// Set the new runtime code.
 		///
-		/// ## Complexity
+		/// # <weight>
 		/// - `O(C + S)` where `C` length of `code` and `S` complexity of `can_set_code`
+		/// - 1 call to `can_set_code`: `O(S)` (calls `sp_io::misc::runtime_version` which is
+		///   expensive).
+		/// - 1 storage write (codec `O(C)`).
+		/// - 1 digest item.
+		/// - 1 event.
+		/// The weight of this function is dependent on the runtime, but generally this is very
+		/// expensive.
+		/// # </weight>
 		#[pallet::call_index(2)]
-		#[pallet::weight((T::BlockWeights::get().max_block, DispatchClass::Operational))]
+		#[pallet::weight((T::SystemWeightInfo::set_code(), DispatchClass::Operational))]
 		pub fn set_code(origin: OriginFor<T>, code: Vec<u8>) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			Self::can_set_code(&code)?;
 			T::OnSetCode::set_code(code)?;
-			Ok(().into())
+			// consume the rest of the block to prevent further transactions
+			Ok(Some(T::BlockWeights::get().max_block).into())
 		}
 
 		/// Set the new runtime code without doing any checks of the given `code`.
 		///
-		/// ## Complexity
+		/// # <weight>
 		/// - `O(C)` where `C` length of `code`
+		/// - 1 storage write (codec `O(C)`).
+		/// - 1 digest item.
+		/// - 1 event.
+		/// The weight of this function is dependent on the runtime.
+		/// # </weight>
```

Comment:
Please remove them. There was a pr that removed them, maybe it didn't yet hit master.